autosubmit/platforms/paramiko_platform.py

```diff
@@ -1045,106 +1080,111 @@ def x11_status_checker(self, session, session_fileno):
                             channel.close()
                             counterpart.close()
                             del self.channels[fd]
-            except Exception as e:
-                pass
 
-    def exec_command(self, command, bufsize=-1, timeout=30, get_pty=False, retries=3, x11=False):
-        """
-        Execute a command on the SSH server.  A new `.Channel` is opened and
-        the requested command is execed.  The command's input and output
-        streams are returned as Python ``file``-like objects representing
-        stdin, stdout, and stderr.
+    def exec_command(
+            self, command: str, bufsize=-1, timeout=30, retries=3, x11=False
+    ) -> tuple[Union[Any, bool], Union[Any, bool], Union[Any, bool]]:
+        """Execute a command on the SSH server.
+
+        A new ``.Channel`` is open and the requested command is executed.
+        The command's input and output streams are returned as Python
+        ``file``-like objects representing stdin, stdout, and stderr.
 
         :param x11:
         :param retries:
-        :param get_pty:
         :param command: the command to execute.
         :type command: str
         :param bufsize: interpreted the same way as by the built-in ``file()`` function in Python.
         :type bufsize: int
-        :param timeout: set command's channel timeout. See `Channel.settimeout`.settimeout.
+        :param timeout: set command's channel timeout. See ``Channel.settimeout``.
         :type timeout: int
         :return: the stdin, stdout, and stderr of the executing command
-
-        :raises SSHException: if the server fails to execute the command
         """
-        while retries > 0:
+        for retry in range(0, retries):
+            Log.debug(f'Executing command {command}, retry #{retry + 1} out of {retries}')
             try:
+                chan: Channel = self.transport.open_session()
+
+                # TODO: x11 appears to never be called anywhere in the code base?
                 if x11:
-                    display = os.getenv('DISPLAY')
-                    if display is None or not display:
-                        display = "localhost:0"
+                    display = os.getenv('DISPLAY', 'localhost:0')
                     try:
                         self.local_x11_display = xlib_connect.get_display(display)
                     except Exception as e:
-                        Log.warning(f"X11 display not found: {e}")
+                        Log.warning(f"X11 display not found: {str(e)}")
                         self.local_x11_display = None
-                    chan = self.transport.open_session()
                     chan.request_x11(single_connection=False, handler=self.x11_handler)
-                else:
-                    chan = self.transport.open_session()
-                if x11:
+
```

Comment:
Above, there's some simplification by using a `range` for retries, and separating/simplifying the logic for x11 (which doesn't appear to be used -- the code looks a bit broken :shrug: )